enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -357,6 +355,17 @@ func UploadTree(ctx context.Context, env environment.Env, dirHelper *DirHelper,
 				if err != nil {
 					return nil, err
 				}
+
+				if _, ok := dirHelper.FindParentOutputPath(fqfn); !ok {
+					actionResult.OutputSymlinks = append(
+						actionResult.OutputSymlinks,
+						&repb.OutputSymlink{
+							Path:   trimPathPrefix(fqfn, rootDir),
+							Target: target,
+						},
+					)
+				}
```

Comment:
The documentation of `output_symlinks` says:

```
// The output paths of the action that are symbolic links to other paths
```

If I'm understanding it correctly, "output paths" is referring to the `output_paths` field. So I don't think `FindParentOutputPath` is the right thing to do here -- instead we only want to set OutputSymlinks for paths that *exactly match* one of the `OutputPaths`.

Example:

```
input:
  - output_paths= [ /foo, /baz ]
action produces:
  /foo (directory)
    /bar (symlink)
  /baz (symlink)
```

I think the expected `OutputSymlinks` should contain just `/baz` since it exactly matches one of the output paths, but I think with this change it will return both `/bar` and `/baz`.

Reply:
Yup, that's sensible.

For better context, I was also able to dump out the requested Command from rules_go:
```
{
  "arguments":  [
    "bazel-out/k8-opt-exec-60723E07/bin/go/tools/builders/go_path/go_path-bin",
    "-manifest",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path~manifest",
    "-out",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path",
    "-mode",
    "link"
  ],
  "outputFiles":  [
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/extra.txt",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/pkg/linux_amd64/example.com/repo/cmd/bin.a",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/pkg/linux_amd64/example.com/repo/pkg/lib.a",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/pkg/linux_amd64/example.com/repo/pkg/lib/transitive.a",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/pkg/linux_amd64/example.com/repo/vendor/example.com/repo2.a",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/pkg/linux_amd64/github.com/bazelbuild/rules_go/go/tools/bzltestutil.a",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/pkg/plan9_arm/example.com/repo/cmd/bin.a",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/cmd/bin/bin.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/data.txt",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/embed_test.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/embedded_src.txt",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/external_test.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/internal_test.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/lib.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/renamed_embedded_src.txt",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/template/index.html.tmpl",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/testdata/testdata.txt",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/pkg/lib/transitive/transitive.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/example.com/repo/vendor/example.com/repo2/vendored.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/github.com/bazelbuild/rules_go/go/tools/bzltestutil/init.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/github.com/bazelbuild/rules_go/go/tools/bzltestutil/lcov.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/github.com/bazelbuild/rules_go/go/tools/bzltestutil/test2json.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/github.com/bazelbuild/rules_go/go/tools/bzltestutil/wrap.go",
    "bazel-out/k8-fastbuild/bin/tests/core/go_path/link_path/src/github.com/bazelbuild/rules_go/go/tools/bzltestutil/xml.go"
  ],
  "platform":  {
    "properties":  [
      {
        "name":  "OSFamily",
        "value":  "Linux"
      },
      {
        "name":  "container-image",
        "value":  "docker://gcr.io/bazel-public/ubuntu1804-bazel-java11@sha256:2d50853a7edbe59a99bc4141d7a03cb1068157b9766077302b46c4ec94eef151"
      }
    ]
  }
}
```

I did not have the time today but tomorrow, I plan to set this up on my Linux laptop and test + iterate from there.

Reply:
nice, are you also able to capture the local ActionResult?

Reply:
Was able to use

```
build:buildbuddy --experimental_remote_grpc_log=./grpc.log
```

in combination with `bb print` to gain very useful insight :D